### PR TITLE
tests(smokehouse): index for static server. print address

### PR DESCRIPTION
**Pull request by connorjclark**
Minor enhancement.

image

### CODE REVIEW

File: lighthouse-cli/test/smokehouse/perf/expectations.js [hunk 1]
```diff
@@ -79,7 +79,7 @@ module.exports = [
               {resourceType: 'font', requestCount: 2, size: '80000±1000'},
               {resourceType: 'script', requestCount: 3, size: '55000±1000'},
               {resourceType: 'image', requestCount: 2, size: '28000±1000'},
-              {resourceType: 'document', requestCount: 1, size: '2100±100'},
+              {resourceType: 'document', requestCount: 1, size: '2200±100'},
```
Comment (connorjclark) [inline lighthouse-cli/test/smokehouse/perf/expectations.js]: went up b/c content-type header is now added for html docs

Comment (patrickhulce) [reply lighthouse-cli/test/smokehouse/perf/expectations.js]: oh right that's easy to forget that header cost is included in these

File: lighthouse-cli/test/fixtures/static-server.js [hunk 2]
```diff
@@ -79,6 +90,8 @@ function requestHandler(request, response) {
       headers['Content-Type'] = 'image/webp';
     } else if (filePath.endsWith('.json')) {
       headers['Content-Type'] = 'application/json';
+    } else if (filePath.endsWith('.html') || filePath === '/') {
+      headers['Content-Type'] = 'text/html';
```
Comment (patrickhulce) [inline lighthouse-cli/test/fixtures/static-server.js]: this feels 100% unrelated to the index part, right? 

why cause conflicts for yourself in https://github.com/GoogleChrome/lighthouse/pull/9542 :)

File: lighthouse-cli/test/fixtures/static-server.js [hunk 3]
```diff
@@ -25,6 +26,16 @@ function requestHandler(request, response) {
   const queryString = requestUrl.search && parseQueryString(requestUrl.search.slice(1));
   let absoluteFilePath = path.join(__dirname, filePath);
 
+  if (filePath === '/') {
+    const fixturePaths = glob.sync('**/*.html', {cwd: __dirname});
+    const html = `
+      <h1>Smoke test fixtures</h1>
```
Comment (patrickhulce) [inline lighthouse-cli/test/fixtures/static-server.js]: could throw `<html>` in there if that's why you did the content type business?

File: lighthouse-cli/test/fixtures/static-server.js [hunk 4]
```diff
@@ -25,6 +26,16 @@ function requestHandler(request, response) {
   const queryString = requestUrl.search && parseQueryString(requestUrl.search.slice(1));
   let absoluteFilePath = path.join(__dirname, filePath);
 
+  if (filePath === '/') {
```
Comment (patrickhulce) [inline lighthouse-cli/test/fixtures/static-server.js]:
```suggestion
  // Create an index page that lists the available test pages.
  if (filePath === '/') {
```

File: lighthouse-cli/test/fixtures/static-server.js [hunk 5]
```diff
@@ -25,6 +26,16 @@ function requestHandler(request, response) {
...
+    const fixturePaths = glob.sync('**/*.html', {cwd: __dirname});
+    const html = `
+      <h1>Smoke test fixtures</h1>
+      ${fixturePaths.map(p => `<a href=${p}>${p}</a>`).join('<br>')}
```
Comment (patrickhulce) [inline lighthouse-cli/test/fixtures/static-server.js]: does this work on windows, like are the paths returned by glob different? 

do we care if it doesn't? probably not

Comment (connorjclark) [reply lighthouse-cli/test/fixtures/static-server.js]: 🤷‍♂

File: lighthouse-cli/test/fixtures/static-server.js [hunk 6]
```diff
@@ -25,6 +26,17 @@ function requestHandler(request, response) {
...
   let absoluteFilePath = path.join(__dirname, filePath);
 
+  if (filePath === '/') {
+    const fixturePaths = glob.sync('**/*.html', {cwd: __dirname});
+    const html = `
+      <html>
+      <h1>Smoke test fixtures</h1>
+      ${fixturePaths.map(p => `<a href=${p}>${p}</a>`).join('<br>')}
+    `;
+    sendResponse(200, html);
```
Comment (brendankenny) [inline lighthouse-cli/test/fixtures/static-server.js]:
```js
const html = `
  <html>
  <h1>Smoke test fixtures</h1>
  ${fixturePaths.map(p => `<a href="${encodeURI(p)}">${escape(p)}</a>`).join('<br>')}
`;
response.writeHead(200, {
  'Content-Security-Policy': `default-src 'none';`,
});
finishResponse(html);
```

let's be safe :)

File: lighthouse-cli/test/fixtures/static-server.js [hunk 7]
```diff
@@ -150,12 +162,15 @@ const serverForOffline = http.createServer(requestHandler);
...
 if (require.main === module) {
   // Start listening
-  serverForOnline.listen(10200, 'localhost');
-  serverForOffline.listen(10503, 'localhost');
+  const onlinePort = 10200;
+  const offlinePort = 10503;
+  serverForOnline.listen(onlinePort, 'localhost');
+  serverForOffline.listen(offlinePort, 'localhost');
+  console.log(`online:  listening on http://localhost:${onlinePort}`);
+  console.log(`offline: listening on http://localhost:${offlinePort}`);
```
Comment (brendankenny) [inline lighthouse-cli/test/fixtures/static-server.js]: there's really nothing offline or online about these, it's just how we use them for one(?) of the smoke tests (we also use them for same- and different-origin in other tests), but I'm not sure it matters what we call them :)

File: lighthouse-cli/test/fixtures/static-server.js (hunk 6 above)
Comment (connorjclark) [reply lighthouse-cli/test/fixtures/static-server.js]: now i'm wondering what sort of js payloads you can inject using just file paths

**Pull request merged**